crates/storage/src/state.rs

```diff
@@ -394,6 +394,69 @@ impl StarknetBlocksTable {
         }
     }
 
+    /// Returns the version of the given block.
+    pub fn get_version(
+        tx: &Transaction<'_>,
+        block: StarknetBlocksBlockId,
+    ) -> anyhow::Result<StarknetVersion> {
+        let mut statement = match block {
+            StarknetBlocksBlockId::Number(_) => tx.prepare(
+                r"
+                    SELECT
+                        version
+                    FROM
+                        starknet_versions
+                    JOIN
+                        starknet_blocks ON starknet_blocks.version_id = starknet_versions.id
+                    WHERE
+                        starknet_blocks.number = ?
```

Comment:
Interesting format.. I wonder if there is some SQL formatter we should consider. Probably not worth it.